src/tactic/monotonicity/lemmas.lean

```diff
@@ -4,70 +4,11 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Simon Hudon
 -/
 import algebra.order.ring
-import data.nat.basic
 import data.set.lattice
-import order.directed
 import tactic.monotonicity.basic
 
 variables {α : Type*}
 
-@[mono]
-lemma mul_mono_nonneg {x y z : α} [ordered_semiring α]
-  (h' : 0 ≤ z)
-  (h : x ≤ y)
-: x * z ≤ y * z :=
-by apply mul_le_mul_of_nonneg_right; assumption
-
-lemma lt_of_mul_lt_mul_neg_right {a b c : α}  [linear_ordered_ring α]
-  (h : a * c < b * c) (hc : c ≤ 0) : b < a :=
-have nhc : -c ≥ 0, from neg_nonneg_of_nonpos hc,
-have h2 : -(b * c) < -(a * c), from neg_lt_neg h,
-have h3 : b * (-c) < a * (-c), from calc
-     b * (-c) = - (b * c)    : by rewrite neg_mul_eq_mul_neg
-          ... < - (a * c)    : h2
-          ... = a * (-c)     : by rewrite neg_mul_eq_mul_neg,
-lt_of_mul_lt_mul_right h3 nhc
-
-@[mono]
-lemma mul_mono_nonpos {x y z : α} [linear_ordered_ring α]
-  (h' : z ≤ 0) (h : y ≤ x) : x * z ≤ y * z :=
-begin
-  classical,
-  by_contradiction h'',
-  revert h,
-  apply not_le_of_lt,
-  apply lt_of_mul_lt_mul_neg_right _ h',
-  apply lt_of_not_ge h''
-end
-
-@[mono]
-lemma nat.sub_mono_left_strict {x y z : ℕ}
-  (h' : z ≤ x)
-  (h : x < y)
-: x - z < y - z :=
-begin
-  have : z ≤ y,
-  { transitivity, assumption, apply le_of_lt h, },
-  apply @nat.lt_of_add_lt_add_left z,
-  rw [add_tsub_cancel_of_le,add_tsub_cancel_of_le];
-    solve_by_elim
-end
-
-@[mono]
-lemma nat.sub_mono_right_strict {x y z : ℕ}
```

Comment:
It looks to me like the lack of generalisation in this lemma is exactly what makes it work in `mono`, causing the new error in bitvec/basic, even though the general version is obviously valuable

Reply:
We might have to wait for #16480 then.

Reply:
Actually that goal at no point involved subtraction and the failure is just because the two goals `ac_mono` generates have been swapped.